control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go

```diff
@@ -71,9 +74,37 @@ func ReconcileNetworkOperator(network *operatorv1.Network, networkType hyperv1.N
 
 	// Setting the management state is required in order to create
 	// this object. We need to create this object before the cno starts
-	// because mutating many of the values (like vxlanport) is not premitted
+	// because mutating many of the values (like vxlanport) is not permitted
 	// after the cno reconciles this operator CR
 	if network.Spec.ManagementState == "" {
 		network.Spec.ManagementState = "Managed"
 	}
 }
+
+func DetectSuboptimalMTU(ctx context.Context, mgmtClient client.Client,
+	guestNetworkOperator *operatorv1.Network, hcp *hyperv1.HostedControlPlane) error {
+	const recommendedMinMTU = uint32(9000)
+	const ovnkOverhead = 200
+	if hcp.Spec.Platform.Type == hyperv1.KubevirtPlatform &&
+		hcp.Spec.Networking.NetworkType == hyperv1.OVNKubernetes &&
+		guestNetworkOperator.Spec.DefaultNetwork.OVNKubernetesConfig != nil &&
+		guestNetworkOperator.Spec.DefaultNetwork.OVNKubernetesConfig.MTU != nil &&
+		*guestNetworkOperator.Spec.DefaultNetwork.OVNKubernetesConfig.MTU < recommendedMinMTU-ovnkOverhead {
+		meta.SetStatusCondition(&hcp.Status.Conditions, metav1.Condition{
```

Comment:
I believe you need to make a copy of the hcp object here.

it is awkward to set this condition from the HCCO when most of the conditions are set within the CPO.

Reply:
if i make a copy of the hcp here, then I would need to propagate it back to the CPO to set the condition on the actual hcp resource, no? wouldn't it complicate the flow?

Reply:
I'm not sure. in general, this is kind of an odd thing we're doing. It woudl be better if the CPO performed this and we wouldn't have to think about the implications of doing this in the HCCO.

do we have any other precedence of how the HCCO attempts to write back to the hcp to look at.

Reply:
yes, there is a precedence for that: HCCO is responsible for destroying the cloud resources, and it is putting the destruction status in the HCP conditions directly:
https://github.com/openshift/hypershift/blob/main/control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go#L1568-L1582

I'm currently not finding any other suitable place to put this logic in.
the CPO (HCP controller) is reconciling kubevirt's CSI driver, kubevirt CCM, and the network operator related resources, we can't put the MTU condition logic there. is it deserve its own top level reconcile function?
Please advise how to proceed